master/internal/proxy/proxy.go

```diff
@@ -51,55 +27,60 @@ type ProxyHTTPAuth func(echo.Context) (done bool, err error)
 
 // Proxy is an actor that proxies requests to registered services.
 type Proxy struct {
-	lock     sync.RWMutex
-	services map[string]*Service
-
 	HTTPAuth ProxyHTTPAuth
+	Services map[string]*Service
+	Syslog   *logrus.Entry
 }
 
-// Receive implements the actor.Actor interface.
-func (p *Proxy) Receive(ctx *actor.Context) error {
-	switch msg := ctx.Message().(type) {
-	case actor.PreStart:
-		p.services = make(map[string]*Service)
-	case Register:
-		if msg.ServiceID == "" {
-			return nil
-		}
-		p.lock.Lock()
```

Comment:
we will need these locks back. or you can use the `mapx` package I added (but could be good practice to not use it).